script/SetCancellers.s.sol

```diff
@@ -0,0 +1,108 @@
+ * @dev Run this script with:
+ *      forge script script/SetCancellers.s.sol:SetCancellers --rpc-url $RPC_URL --broadcast --verify
+ */
+contract SetCancellers is Script {
+    address constant GOVERNOR_ADDRESS = 0x0000000000000000000000000000000000000000; // Replace with actual Optimism Governor address
+    address constant L2_SAFE_1 = address(0); // Replace with first L2 Safe address
+    address constant L2_SAFE_2 = address(0); // Replace with second L2 Safe address
+    address constant L2_SAFE_3 = address(0); // Replace with third L2 Safe address
+
+    // Constants
+    bytes32 constant PROPOSER_ROLE = keccak256("PROPOSER_ROLE");
+    bytes32 constant CANCELLER_ROLE = keccak256("CANCELLER_ROLE");
+    bytes32 constant TIMELOCK_ADMIN_ROLE = keccak256("TIMELOCK_ADMIN_ROLE");
+
+    function run() external {
+        // Boot deployer private key
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+        vm.startBroadcast(deployerPrivateKey);
+
+        // Get the governor contract
+        OptimismGovernor governor = OptimismGovernor(payable(GOVERNOR_ADDRESS));
+
+        // Get the timelock address from the governor
+        address timelockAddress = governor.timelock();
+        console.log("Timelock address:", timelockAddress);
+
+        // Get the timelock contract
+        TimelockControllerUpgradeable timelock = TimelockControllerUpgradeable(payable(timelockAddress));
+
+        // Get the access control contract
+        AccessControlEnumerableUpgradeable accessControl = AccessControlEnumerableUpgradeable(payable(timelockAddress));
+
+        // Make sure caller has admin role as we're about to modify the timelock
+        address caller = vm.addr(deployerPrivateKey);
+        require(timelock.hasRole(TIMELOCK_ADMIN_ROLE, caller), "Caller does not have admin role on timelock");
+
+        // Step 1: Grant CANCELLER_ROLE to the three L2 Safes (if they don't already have it)
+        if (!timelock.hasRole(CANCELLER_ROLE, L2_SAFE_1)) {
+            console.log("Granting CANCELLER_ROLE to L2 Safe 1:", L2_SAFE_1);
+            timelock.grantRole(CANCELLER_ROLE, L2_SAFE_1);
+        } else {
+            console.log("L2 Safe 1 already has CANCELLER_ROLE:", L2_SAFE_1);
+        }
+
+        if (!timelock.hasRole(CANCELLER_ROLE, L2_SAFE_2)) {
+            console.log("Granting CANCELLER_ROLE to L2 Safe 2:", L2_SAFE_2);
+            timelock.grantRole(CANCELLER_ROLE, L2_SAFE_2);
+        } else {
+            console.log("L2 Safe 2 already has CANCELLER_ROLE:", L2_SAFE_2);
+        }
+
+        if (!timelock.hasRole(CANCELLER_ROLE, L2_SAFE_3)) {
+            console.log("Granting CANCELLER_ROLE to L2 Safe 3:", L2_SAFE_3);
+            timelock.grantRole(CANCELLER_ROLE, L2_SAFE_3);
+        } else {
+            console.log("L2 Safe 3 already has CANCELLER_ROLE:", L2_SAFE_3);
+        }
+
+        // Step 2: Ensure all proposers have CANCELLER_ROLE (which is the default in OpenZeppelin's TimelockController)
```

Comment:
is there a specific reason you check for all proposers and not just the 3 Safe addresses?